File: backend/src/main/java/com/leadgrowth/controller/CallController.java
package com.leadgrowth.controller;

import com.leadgrowth.dto.CallAnalyticsDto;
import com.leadgrowth.dto.CallSessionDto;
import com.leadgrowth.service.CallService;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/calls")
public class CallController {

    private final CallService callService;

    public CallController(CallService callService) {
        this.callService = callService;
    }

    @PostMapping("/start")
    public ResponseEntity<CallSessionDto> startCall(@RequestBody Map<String, Object> payload) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        Long leadId = Long.parseLong(String.valueOf(payload.get("leadId")));
        return ResponseEntity.ok(callService.startCall(leadId, email));
    }

    @PostMapping("/end")
    public ResponseEntity<CallSessionDto> endCall(@RequestBody(required = false) Map<String, Object> payload) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        Long callId = (payload != null && payload.get("callId") != null) ? Long.parseLong(String.valueOf(payload.get("callId"))) : null;
        String notes = (payload != null && payload.get("notes") != null) ? String.valueOf(payload.get("notes")) : "";
        return ResponseEntity.ok(callService.endCall(callId, email, notes));
    }

    @GetMapping("/active")
    public ResponseEntity<CallSessionDto> getActiveCall() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getActiveCall(email));
    }

    @GetMapping("/history/{leadId}")
    public ResponseEntity<List<CallSessionDto>> getCallHistoryForLead(@PathVariable Long leadId) {
        return ResponseEntity.ok(callService.getCallHistoryForLead(leadId));
    }

    @GetMapping("/user")
    public ResponseEntity<CallAnalyticsDto> getUserCallAnalytics() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getUserCallAnalytics(email));
    }

    @GetMapping("/team")
    public ResponseEntity<CallAnalyticsDto> getTeamCallAnalytics() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getTeamCallAnalytics(email));
    }

    @GetMapping("/dashboard")
    public ResponseEntity<CallAnalyticsDto> getDashboardCallAnalytics() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getUserCallAnalytics(email));
    }

    @GetMapping("/analytics")
    public ResponseEntity<CallAnalyticsDto> getAnalytics() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getTeamCallAnalytics(email));
    }

    @GetMapping("/reports")
    public ResponseEntity<List<CallSessionDto>> getCallReports(
            @RequestParam(required = false) Long userId,
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate
    ) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(callService.getCallReports(email, userId, startDate, endDate));
    }
}

File: backend-dotnet/Controllers/CallController.cs
using System.Security.Claims;
using LeadGrowth.DTOs;
using LeadGrowth.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace LeadGrowth.Controllers;

[ApiController]
[Route("api/calls")]
[Authorize]
public class CallController : ControllerBase
{
    private readonly ICallService _callService;

    public CallController(ICallService callService)
    {
        _callService = callService;
    }

    [HttpPost("start")]
    public async Task<ActionResult<CallSessionDto>> StartCall([FromBody] Dictionary<string, object> payload)
    {
        var email = GetUserEmail();
        long leadId = long.Parse(payload["leadId"].ToString()!);
        try
        {
            var session = await _callService.StartCallAsync(leadId, email);
            return Ok(session);
        }
        catch (ArgumentException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpPost("end")]
    public async Task<ActionResult<CallSessionDto>> EndCall([FromBody] Dictionary<string, object>? payload)
    {
        var email = GetUserEmail();
        long? callId = payload != null && payload.ContainsKey("callId") && payload["callId"] != null ? long.Parse(payload["callId"].ToString()!) : null;
        string? notes = payload != null && payload.ContainsKey("notes") && payload["notes"] != null ? payload["notes"].ToString() : "";

        try
        {
            var session = await _callService.EndCallAsync(callId, email, notes);
            return Ok(session);
        }
        catch (ArgumentException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpGet("active")]
    public async Task<ActionResult<CallSessionDto?>> GetActiveCall()
    {
        var email = GetUserEmail();
        var session = await _callService.GetActiveCallAsync(email);
        return Ok(session);
    }

    [HttpGet("history/{leadId}")]
    public async Task<ActionResult<List<CallSessionDto>>> GetCallHistoryForLead(long leadId)
    {
        var list = await _callService.GetCallHistoryForLeadAsync(leadId);
        return Ok(list);
    }

    [HttpGet("user")]
    public async Task<ActionResult<CallAnalyticsDto>> GetUserCallAnalytics()
    {
        var email = GetUserEmail();
        var analytics = await _callService.GetUserCallAnalyticsAsync(email);
        return Ok(analytics);
    }

    [HttpGet("team")]
    public async Task<ActionResult<CallAnalyticsDto>> GetTeamCallAnalytics()
    {
        var email = GetUserEmail();
        var analytics = await _callService.GetTeamCallAnalyticsAsync(email);
        return Ok(analytics);
    }

    [HttpGet("dashboard")]
    public async Task<ActionResult<CallAnalyticsDto>> GetDashboardCallAnalytics()
    {
        return await GetUserCallAnalytics();
    }

    [HttpGet("analytics")]
    public async Task<ActionResult<CallAnalyticsDto>> GetAnalytics()
    {
        return await GetTeamCallAnalytics();
    }

    [HttpGet("reports")]
    public async Task<ActionResult<List<CallSessionDto>>> GetCallReports([FromQuery] long? userId, [FromQuery] string? startDate, [FromQuery] string? endDate)
    {
        var email = GetUserEmail();
        var list = await _callService.GetCallReportsAsync(email, userId, startDate, endDate);
        return Ok(list);
    }

    private string GetUserEmail()
    {
        return User.FindFirstValue(ClaimTypes.Name) ?? User.FindFirstValue(ClaimTypes.Email) ?? string.Empty;
    }
}
